### PR TITLE
fix(integ-runner): Fix call to spawnSync for hooks commands

### PR DESCRIPTION
**Pull request by arewa**
Fixes the issue #22344

I reworked the approach of calling `exec` by splitting each command in hook to the command itself and it's arguments. All hooks were affected: preDeploy, postDeploy, preDestroy, postDestroy.

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/integ-tests/test/integ.app.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { App, Stack } from '@aws-cdk/core';
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/integ-tests/test/integ.app.ts:1]: This isn't testing anything. Why is this here?

File: packages/@aws-cdk/integ-tests/package.json [hunk 2]
```diff
@@ -52,7 +52,8 @@
     "compat": "cdk-compat",
     "rosetta:extract": "yarn --silent jsii-rosetta extract",
     "build+extract": "yarn build && yarn rosetta:extract",
-    "build+test+extract": "yarn build+test && yarn rosetta:extract"
+    "build+test+extract": "yarn build+test && yarn rosetta:extract",
+    "integ": "integ-runner"
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/integ-tests/package.json:56]: This isn't needed in this package.

File: packages/@aws-cdk/integ-tests/package.json [hunk 3]
```diff
@@ -65,6 +66,7 @@
     "@aws-cdk/cdk-build-tools": "0.0.0",
     "@aws-cdk/cx-api": "0.0.0",
     "@aws-cdk/pkglint": "0.0.0",
+    "@aws-cdk/integ-runner": "0.0.0",
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/integ-tests/package.json:69]: Nor this.

**Review by TheRealAmazonKendra — DISMISSED** [review]

File: packages/@aws-cdk/integ-tests/test/integ.app.ts (hunk 1 above)
Comment (arewa) [reply packages/@aws-cdk/integ-tests/test/integ.app.ts:1]: @TheRealAmazonKendra, I can actually remove this I added it to avoid the linter issue saying PR must have at least one integration test update. But integ-runner doesn't have its own integration tests that's why I added this small test. And with this test `integ-tests/package.json` will also be reverted.

File: packages/@aws-cdk/integ-runner/lib/utils.ts [hunk 4]
```diff
@@ -41,6 +41,18 @@ export function chain(commands: string[]): string {
...
 }
 
+/**
+ * Prepare command for execution by exec
+ */
+export function prepareForExec(command: string): string[] {
+  let chunks = command.split(/\s+/);
+  const cmd = chunks.shift();
+  if (cmd !== undefined) {
+    return [cmd, chunks.join(' ')];
```
Comment (corymhall) [inline packages/@aws-cdk/integ-runner/lib/utils.ts:51]: Should this be `return chunks` instead?

For example if I had 

```
hooks: {
  preDeploy: ['npx cdk deploy'],
}
```

I would want it to return `['npx', 'cdk', 'deploy']` right?

**Review by corymhall — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/integ-runner/lib/utils.ts (hunk 4 above)
Comment (arewa) [reply packages/@aws-cdk/integ-runner/lib/utils.ts:51]: @corymhall, thank you for the comment, for the command you mentioned it could be like that. But for example for the command `echo "preDeploy hook"` if we just split it by space it will result in `['echo', '"preDeploy', 'hook"'] `which seems to be not the correct input for `exec`.

Comment (corymhall) [reply packages/@aws-cdk/integ-runner/lib/utils.ts:51]: You could do something like this instead in order to keep quoted sections together.

```ts
let chunks = command.match(/(?:[^\s"]+|"[^"]*")+/g)
// ['echo', '"preDeploy hook"']
```

I think if you use arguments with spawnSync each argument to the command needs to be its own element in the list.

Comment (arewa) [reply packages/@aws-cdk/integ-runner/lib/utils.ts]: Yes, make sense, I updated the solution. Thank you again for the review.

File: packages/@aws-cdk/integ-runner/lib/utils.ts [hunk 5]
```diff
@@ -41,6 +41,17 @@ export function chain(commands: string[]): string {
   return commands.filter(c => !!c).join(' && ');
 }
 
+/**
+ * Split command to chunks by space
+ */
+export function chunks(command: string): string[] {
+  const result = command.match(/(?:[^\s"]+|"[^"]*")+/g);
+  if (result !== null) {
```
Comment (corymhall) [inline packages/@aws-cdk/integ-runner/lib/utils.ts:49]: nit

```ts
return result ?? [];
```

**Review by corymhall — CHANGES_REQUESTED** [review]
Looks great! just one minor nit and then we are good to go!

**Review by corymhall — APPROVED** [review]
